internal/tasks/fleet_validate.go

```diff
@@ -113,8 +116,13 @@ func (t *FleetValidateLogic) validateFleetTemplate(ctx context.Context, fleet *a
 	} else {
 		condition.Status = api.ConditionStatusFalse
 		condition.Reason = util.StrToPtr("Invalid")
-		condition.Message = util.StrToPtr(fmt.Sprintf("Fleet has %d invalid configurations: %s", len(invalidConfigs), strings.Join(invalidConfigs, ", ")))
-		retErr = fmt.Errorf("found %d invalid configurations: %s", len(invalidConfigs), strings.Join(invalidConfigs, ", "))
+		configurationStr := "configuration"
+		if len(invalidConfigs) > 1 {
+			configurationStr += "s"
+		}
+		msg := fmt.Sprintf("Fleet has %d invalid %s: %s", len(invalidConfigs), configurationStr, strings.Join(invalidConfigs, ", "))
+		condition.Message = util.StrToPtr(msg)
+		retErr = fmt.Errorf(msg)
```

Comment:
slightly cleaner using Errorf directly
```suggestion
retErr = fmt.Errorf("fleet has %d invalid %s: %s", len(invalidConfigs), configurationStr, strings.Join(invalidConfigs, ", "))
condition.Message = util.StrToPtr(err.Error())
```